dms_field/tests/test_dms_field.py

```diff
@@ -244,3 +244,40 @@ def test_parents(self):
             {"id": directory.id, "name": directory.name},
             directory.search_read_parents(fields=["id", "name"]),
         )
+
+    def test_child_values(self):
+        """Values of the child directory in the template
+        are propagated to the new directories."""
+        # Arrange
+        partner = self.partner
+        access_group = self.env["dms.access.group"].create(
+            {
+                "name": "Test Access group",
+            }
+        )
+        child_dir = self.subdirectory_1
+        child_dir.inherit_group_ids = False
+        child_dir.group_ids = access_group
+
+        # Act
+        self.env["dms.field.template"].with_context(
+            res_model=partner._name,
+            res_id=partner.id,
+        ).create_dms_directory()
+
+        # Assert
+        partner.invalidate_model()
+        new_child_dir = partner.dms_directory_ids.child_directory_ids.filtered_domain(
+            [
+                ("name", "=", child_dir.name),
+            ]
+        )
+        self.assertRecordValues(
+            new_child_dir,
+            [
+                {
+                    "inherit_group_ids": child_dir.inherit_group_ids,
+                    "group_ids": child_dir.group_ids.ids,
+                }
+            ],
+        )
```

Comment:
```suggestion
        subdirectory_1 = partner.dms_directory_ids.child_directory_ids.filtered(
            lambda x: x.name == self.subdirectory_1.name
        )
        self.assertFalse(subdirectory_1.inherit_group_ids)
        self.assertEqual(subdirectory_1.group_ids, access_group)
        subdirectory_2 = partner.dms_directory_ids.child_directory_ids.filtered(
            lambda x: x.name == self.subdirectory_2.name
        )
        self.assertTrue(subdirectory_2.inherit_group_ids)
        self.assertIn(self.group, subdirectory_2.group_ids)
```
IMO this is more readable and checks that both subdirectories have the right data.

Reply:
Thanks for the quick review!
This needed some tweaking, but I accepted your suggestion overall, please check